plan/new_physical_plan_builder.go

```diff
@@ -282,22 +280,89 @@ func (p *LogicalJoin) generatePhysicalPlans() []PhysicalPlan {
 	}
 }
 
-func (p *LogicalJoin) preferUseMergeJoin() bool {
-	return p.preferMergeJoin && len(p.EqualConditions) == 1
+func getPermutation(cols1, cols2 []*expression.Column) ([]int, []*expression.Column) {
+	tmpSchema := expression.NewSchema(cols2...)
+	permutation := make([]int, 0, len(cols1))
+	for i, col1 := range cols1 {
+		offset := tmpSchema.ColumnIndex(col1)
+		if offset == -1 {
+			return permutation, cols1[:i]
+		}
+		permutation = append(permutation, offset)
+	}
+	return permutation, cols1
```

Comment:
If all column can match cols1, we return cols1. If part of columns can match cols1, we return part of cols1 in line 289